docs/tools/bootloader.md

```diff
@@ -35,12 +38,58 @@ This parameter defines the bootloader image to be used during the built-in postb
+    - `16le` A value represented as 16 bits little endian.
+    - `32le` A value represented as 32 bits little endian.
+    - `64le` A value represented as 64 bits little endian.
+
+- `timestamp` A time stamp value in seconds from epoch in the timezone GMT/UTC. The argument to this type is always null.
+    - `64be` A time stamp value truncated to 64 bits big endian.
+    - `64le` A time stamp value truncated to 64 bits little endian.
+
+- `digest` A digest of an image. All digests are computed in the order they appear. A digest of the header digests the header up to its start address.
+    - `CRCITT32be` A big endian 32-bit checksum of an image with an initial value of `0xffffffff`, input reversed, and output reflected.
+    - `CRCITT32le` A little endian 32bit checksum of an image with an initial value of `0xffffffff`, input reversed, and output reflected.
+    - `SHA256` A SHA-2 using a block-size of 256 bits.
+    - `SHA512` A SHA-2 using a block-size of 512 bits.
+
+- `size` The size of a list of images, added together.
+    - `32be` A size represented as 32 bits big endian.
+    - `64be` A size represented as 64 bits big endain.
+    - `32le` A size represented as 32 bits little endian.
+    - `64le` A size represented as 64 bits little endain.
+
+Items in the application header are packed starting where the previous field ended; items in the header are packed using the C "packed" struct semantics. The presence of the `target.header_format` format field defines two macros `MBED_HEADER_START`, which expands to the start address of the firmware header, and `MBED_HEADER_SIZE`, containing the size in bytes of the firmware header.The region following the application header starts at `MBED_HEADER_START + MBED_HEADER_SIZE` rounded up to a multiple of 8 bytes.
+
+You may use this parameter in conjunction with `target.bootloader_img`, `target.restrict_size`, `target.header_offset` and `target.app_offset`. It conflicts with `target.mbed_app_start` and `target.mbed_app_size`. When used with `target.bootloader_img`, that parameter defines the start of the application. Otherwise, the start is the start of ROM.
+
+### `target.header_offset`
+
+This parameter directly assigns the offset of the beginning of the header section defined in `target.header_format`. This parameter creates space between the bootloader and application header or asserts that the bootloader is at most as big as the specified offset.
+
+You may use this parameter in conjunction with `target.bootloader_img`, `target.restrict_size`, `target.header_format` and `target.app_offset`. It conflicts with `target.mbed_app_start` and `target.mbed_app_size`.
+
+### `target.app_offset`
+
+This parameter directly assigns the offset of the beginning of the application section that follows the header, when defined in `target.header_format`, or the bootloader defined in `target.bootloader_img`. This parameter creates space between the application header and the application.
```

Comment:
Is this saying that this parameter assigns the offset of the application after the header when the header is defined in target.header_format or the bootloader is defined in target.bootloader_imag? Or something else?

Reply:
Not quite. 
It's more like:
If there's a header, the application follows the header. Otherwise the application follows the bootloader. If `target.app_offset` parameter is present, it changes the application offset.

I suppose I don't need to specify the application location here.